R/zchunk_L102.GDP.R

```diff
@@ -141,18 +143,19 @@ module_socioeconomics_L102.GDP <- function(command, ...) {
     ## Add region IDs to historical GDP and aggregate by region. Hang onto the country-level data
     ## because we will use them again when we make the IMF adjustments
     gdp_mil90usd_ctry <-
-      left_join_error_no_match(L100.gdp_mil90usd_ctry_Yh, iso_region32_lookup,
-                               by = 'iso') %>%
+      left_join_error_no_match(L100.gdp_mil90usd_ctry_Yh, iso_region32_lookup, by = 'iso') %>%
       rename(gdp = value)
+
     gdp_mil90usd_rgn <- gdp_mil90usd_ctry %>%
+      filter(year %in% HISTORICAL_YEARS) %>%
```

Comment:
Interesting.  I don't think the code should be responsible for validating those choices, but if there's a requirement, then we should at least document it for users.  For example, as it stands, I'd have to go spelunking in the code to tell you whether `HISTORICAL_YEARS` and `FUTURE_YEARS` are supposed to overlap or not allowed to.